src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -159,26 +159,17 @@ public static void JsonIgnoreAttribute_UnsupportedCollection()
             
             // Using new options instance to prevent using previously cached metadata.
             JsonSerializerOptions options = new JsonSerializerOptions();
-            string serialized = JsonSerializer.Serialize(new ClassWithUnsupportedDictionary(), options);
-            
-            // Object keys are fine on serialization if the keys are strings.
-            Assert.Contains(@"""MyConcurrentDict"":null", serialized);
-            Assert.Contains(@"""MyIDict"":null", serialized);
-            Assert.Contains(@"""MyDict"":null", serialized);
+
+            // Unsupported collections will throw on serialize by default.
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize(new ClassWithUnsupportedDictionary(), options));
```

Comment:
PTAL @layomia

Reply:
Yes, the previous output was wrong. Generic dictionaries should explicitly have typeof(string) as the key. I think this can be revisited when supporting non-string keys for dictionaries, cc. @Jozkee